source/common/http/http2/codec_impl.cc

```diff
@@ -118,6 +122,40 @@ void ConnectionImpl::StreamImpl::encodeTrailers(const HeaderMap& trailers) {
     parent_.sendPendingFrames();
   }
 }
+void ConnectionImpl::StreamImpl::readDisable(bool disable) {
+  ENVOY_CONN_LOG(debug, "Stream {} disabled: disable {}, unconsumed_bytes {}", parent_.connection_,
+                 stream_id_, disable, unconsumed_bytes_);
+  if (disable) {
+    ++buffers_overrun_;
+  } else {
+    --buffers_overrun_;
+    if (!buffers_overrun()) {
+      nghttp2_session_consume(parent_.session_, stream_id_, unconsumed_bytes_);
+      unconsumed_bytes_ = 0;
+      parent_.sendPendingFrames();
+    }
+  }
+}
+
+void ConnectionImpl::StreamImpl::pendingRecvBufferHighWatermark() {
```

Comment:
Given the work being factored into the watermark classes I don't think it gains us much but I'm happy to err on the side of caution.  Added locally, upstreamed soon :-)